src/Install/DatabaseConfig.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * (c) Toby Zerner <toby.zerner@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Flarum\Install;
+
+class DatabaseConfig
+{
+    private $driver;
+    private $host;
+    private $port;
+    private $database;
+    private $username;
+    private $password;
+    private $prefix;
+
+    public function __construct($driver, $host, $port, $database, $username, $password, $prefix)
+    {
+        $this->driver = $driver;
+        $this->host = $host;
+        $this->port = $port;
+        $this->database = $database;
+        $this->username = $username;
+        $this->password = $password;
+        $this->prefix = $prefix;
+
+        $this->validate();
+    }
+
+    public function getConfig(): array
```

Comment:
Just curious whether making this an `Arrayable` or `__invoke` kind of class would be cleaner?

Arrayable makes `toArray` obligatory: https://laravel.com/api/5.7/Illuminate/Contracts/Support/Arrayable.html

Reply:
If this were a PHP core interface like [`JsonSerializable`](http://php.net/manual/en/class.jsonserializable.php), maybe this would even be called implicitly in some cases. It isn't, though. :confused: 

Still, I'll follow your suggestion - at least this way I don't have to come up with my own semi-good method name. :laughing: